base/asyncmap.jl

```diff
@@ -175,7 +175,19 @@ function maptwice(wrapped_f, chnl, worker_tasks, c...)
     close(chnl)
 
     # check and throw any exceptions from the worker tasks
-    foreach(x->(v=fetch(x); isa(v, Exception) && throw(v)), worker_tasks)
+    exs = nothing
+    for t in worker_tasks
+        try
+            fetch(t)
+        catch ex
+            if exs == nothing
+                exs = [ex]
+            else
+                push!(exs, ex)
+            end
+        end
+    end
+    exs != nothing && throw(CompositeException(exs))
```

Comment:
This loop is functionality almost exactly the same as a `@sync` block with the new macro `@sync_add` from #32814, but `@sync calls `wait` rather than `fetch`. Would the following work?

```julia
@sync begin
    foreach(t->@sync_add(t), worker_tasks)
end
```

(or just `sync_end(worker_tasks)`, though that couples this to the implementation detail in task.jl)